lib/xcodeproj/project/object/build_configuration.rb

```diff
@@ -103,6 +104,18 @@ def expand_build_setting(build_setting_value, config_value)
           build_setting_value.map { |value| Constants::INHERITED_KEYWORDS.include?(value) ? inherited : value }.flatten
         end
 
+        def resolve_variable_substitution(config_setting)
+          expression = /\$\((.*)\)/
+          match_data = config_setting.match(expression)
+          if match_data.nil?
+            return name if config_setting.eql?('CONFIGURATION')
```

Comment:
Not sure if this is an exhaustive way to resolve the value, would love feedback on this.